flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java

```diff
@@ -152,33 +154,32 @@ void runDetachedModeTest(Map<String, String> securityProperties) throws Exceptio
 
         jobRunner.join();
 
+        final long timeoutInSecs = 20;
+        final long testConditionIntervalInMillis = 500;
         // in "new" mode we can only wait after the job is submitted, because TMs
         // are spun up lazily
-        LOG.info("Waiting until two containers are running");
         // wait until two containers are running
-        while (getRunningContainers() < 2) {
-            sleep(500);
-        }
+        LOG.info("Waiting until two containers are running");
+        CommonTestUtils.waitUntilCondition(
+                () -> getRunningContainers() == 2,
+                Deadline.fromNow(Duration.ofSeconds(timeoutInSecs)),
```

Comment:
we may want to use a larger timeout here; the previous version was effectively waiting forever. The yarn tests are tediously slow, so several minutes might be required

Reply:
It was implicitly set through the test's timeout? Do we have a rule of thumb for timeouts? Rather have it on the call than the test? I increased the per-call timeout to 1 minute and removed the overall timeout of 1 minute. This would give the test an overall timeout of 3 minutes now.

Reply:
We don't have a proper pattern for timeouts. Test-scoped timeouts don't quite fulfill the same role as call-based timeouts, since they also catch things that aren't immediately obvious, like blocking calls. I don't think we should remove the test timeout for the time being.

Reply:
Good point. I'm gonna revert the test-scoped timeout removal